src/LoginWithAuth0.ts

```diff
@@ -0,0 +1,91 @@
+import { Auth0Client } from "@auth0/auth0-spa-js"
+
+const DOMAIN_SUFFIX = "wandelbots.io"
+
+/** Mapping of stages to Auth0 configurations */
+const auth0ConfigMap = {
+  dev: {
+    domain: `https://auth.portal.dev.${DOMAIN_SUFFIX}`,
+    clientId: process.env.NOVA_AUTH0_DEV_CLIENT_ID,
+  },
+  stg: {
+    domain: `https://auth.portal.stg.${DOMAIN_SUFFIX}`,
+    clientId: process.env.NOVA_AUTH0_STG_CLIENT_ID,
+  },
+  prod: {
+    domain: `https://auth.portal.${DOMAIN_SUFFIX}`,
+    clientId: process.env.NOVA_AUTH0_PROD_CLIENT_ID,
+  },
+}
+
+/** Determine which Auth0 configuration to use based on instance URL  */
+const getAuth0Config = (instanceUrl: string) => {
+  if (instanceUrl.includes(`dev.${DOMAIN_SUFFIX}`)) return auth0ConfigMap.dev
+  if (instanceUrl.includes(`stg.${DOMAIN_SUFFIX}`)) return auth0ConfigMap.stg
+  if (instanceUrl.includes(DOMAIN_SUFFIX)) return auth0ConfigMap.prod
+  throw new Error(
+    "Unsupported instance URL. Cannot determine Auth0 configuration.",
+  )
+}
+
+/**
+ * Checks if login is required based on the instance URL.
+ */
+export const isLoginRequired = (instanceUrl: string): boolean => {
+  return instanceUrl.includes(DOMAIN_SUFFIX)
+}
+
+export const isDeployedOnInstance = (): boolean => {
```

Comment:
We might want to call this `isDeployedOnPortalInstance` since "instance" is a bit ambiguous (you can have an "instance" from wabo.run as well)